orchagent/neighorch.cpp

```diff
@@ -348,8 +348,10 @@ void NeighOrch::doTask(Consumer &consumer)
 
             if (m_syncdNeighbors.find(neighbor_entry) == m_syncdNeighbors.end() || m_syncdNeighbors[neighbor_entry] != mac_address)
             {
-                if (addNeighbor(neighbor_entry, mac_address))
+                if (addNeighbor(neighbor_entry, mac_address)) {
+                    gPortsOrch->increasePortNeighRefCount(p.m_alias);
```

Comment:
I think, at-least in L3 case, neighbor is dependent on interface and interface is dependent on port. Also, we can claim that "if we have a L3 neighbor entry depending on port, we will have a corresponding interface entry depending on port" Hence let the port handle the dependency! In L2 neighbor case I need to think, do we create an L2 interface?